src/yio-plugin/integration.h

```diff
@@ -55,31 +55,34 @@ class Integration : public QObject, public IntegrationInterface {
                 YioAPIInterface* api, ConfigInterface* configObj, Plugin* plugin);
     explicit Integration(Plugin* plugin);
 
-    ~Integration();
+    ~Integration() override;
 
     Q_PROPERTY(int state READ state NOTIFY stateChanged)
 
-    Q_INVOKABLE void connect()    = 0;                                             // Must be implemented by integration
-    Q_INVOKABLE void disconnect() = 0;                                             // Must be implemented by integration
-    void             enterStandby() {}                                             // Can be overriden by integration
-    void             leaveStandby() {}                                             // Can be overriden by integration
-    QStringList      getAllAvailableEntities() { return m_allAvailableEntities; }  // Can be overriden by integration
-    Q_INVOKABLE void sendCommand(const QString& type, const QString& entity_id, int command, const QVariant& param) = 0;
+    Q_INVOKABLE void connect() override    = 0;  // Must be implemented by integration
+    Q_INVOKABLE void disconnect() override = 0;  // Must be implemented by integration
```

Comment:
I don't see the reason for Q_INVOKABLE. Everything should be handled by C++ now afaik. Therefore defining them as slots would be better, or I'm still missing something.
Or do we really allow people to write integration plugins in QML?

Reply:
connect and disconnect is `Q_INVOKABLE`, so you can manually disconnect the integration from the UI. If you go to `Settings/Integrations` there is option to connect/disconnect.